### DISCUSSION TITLE
Running Docker (and Docker Compose) inside WSL?

### DISCUSSION THREAD

**Discussion by Luc45** [Vampire/setup-wsl#53] [Q&A] [answered]
Has anyone had any luck running Docker and Docker Compose inside WSL in GitHub Actions?

**Comment by Vampire** [comment]
As only WSL 1 is available, I don't think you can. As far as I remember it was only possible to use the Docker CLI inside WSL 1 to control a Docker daemon actually running on Windows.

**Comment by Vampire** [comment, marked as answer]
Just in case you missed it, since some time also WSLv2 is possible and also supported by this action, so now you should be able to use Docker inside WSL in GHA. :-)

**Reply by Luc45** [reply]
Thanks, @Vampire!

**Reply by pseusys** [reply]
Do you by any chance have an example of setting up docker inside of the action?

**Reply by Vampire** [reply]
Just follow the normal docker installation instructions?

**Reply by pseusys** [reply]
Yes, thank you. After all (in case it could be useful for anyone following my path in the future), I think that for less complex projects it could be really useful just using `podman` as a replacement, since it does not require any complex installation: just adding `podman`, `podman-docker` and `podman-compose` in `additional-packages` seems to be enough.

**Reply by Vampire** [reply]
Nowadays I would practically always use podman over docker.
It should to all relevant parts a drop-in replacement and should not be restricted by project complexity.

Having said that, installing docker should also not be more complex, depending on which distribution you use.

For example with Debian or Ubuntu you would use the `docker.io` package.

**Comment by pseusys** [comment]
Hello @Vampire!
Thank you for your great action, it is very helpful! However, for now unfortunately I couldn't use it for one of my projects because of Docker setup. I use the Docker convenience script (`curl -fsSL https://get.docker.com | sh`) and for some reason I think that Docker is not getting installed correctly: when I start my docker compose project in WSL all the bridge interfaces appear to be `DOWN` (as well as `docker0` itself) - and all the routes to my containers are also `linkdown`, meaning containers can not be accessed. I have never faced an issue like that neither on my Linux or Windows WSL machines. If you (or any other action users) have ever succeeded in running a docker compose project with bridge networks in WSL - could you please share your setup? Thank you a lot in advance!

**Reply by Vampire** [reply]
Didn't you say you use podman now instead and it works? 

I never used docker inside WSL on GHA so far, sorry.

Maybe it works better if you do not use the script install method, but the OS packages, as I described above. 🤷‍♂️

**Reply by pseusys** [reply]
I tried all of the setups. For one container setup podman works fine, but then I went further to launch a multiple-network and multiple-container project - and failed... let me know if you have any more updates. Bridge networks just seem fail to start.

**Reply by Vampire** [reply]
I recommend you ask in some docker community instead.
